docs/settings.rst
.. _settings:

###############
Global Settings
###############

* BLOG_IMAGE_THUMBNAIL_SIZE: Size of the main image when shown on the post
  lists; it's a dictionary with ``size``, ``crop`` and ``upscale`` keys;
  (default: ``{'size': '120x120', 'crop': True,'upscale': False}``)
* BLOG_IMAGE_FULL_SIZE: Size of the main image when shown on the post
  detail; it's a dictionary with ``size``, ``crop`` and ``upscale`` keys;
  (default: ``{'size': '640x120', 'crop': True,'upscale': False}``)
* BLOG_PAGINATION: Number of post per page; (default: ``10``)
* BLOG_MENU_EMPTY_CATEGORIES: Flag to show / hide categories without posts
  attached from the menu; (default: ``True``)
* BLOG_LATEST_POSTS: Default number of post in the **Latest post** plugin;
  (default: ``5``)
* BLOG_POSTS_LIST_TRUNCWORDS_COUNT: Default number of words shown for
  abstract in the post list; (default: ``100``)
* BLOG_TYPE: Generic type for the post object; (default: ``Article``)
* BLOG_TYPES: Choices of available blog types;
  (default: to ``META_OBJECT_TYPES`` defined in `django-meta settings`_)
* BLOG_FB_TYPE: Open Graph type for the post object; (default: ``Article``)
* BLOG_FB_TYPES: Choices of available blog types;
  (default: to ``META_FB_TYPES`` defined in `django-meta settings`_)
* BLOG_FB_APPID: Facebook Application ID
* BLOG_FB_PROFILE_ID: Facebook profile ID of the post author
* BLOG_FB_PUBLISHER: Facebook URL of the blog publisher
* BLOG_FB_AUTHOR_URL: Facebook profile URL of the post author
* BLOG_FB_AUTHOR: Facebook profile URL of the post author
* BLOG_TWITTER_TYPE: Twitter Card type for the post object;
  (default: ``Summary``)
* BLOG_TWITTER_TYPES: Choices of available blog types for twitter;
  (default: to ``META_TWITTER_TYPES`` defined in `django-meta settings`_)
* BLOG_TWITTER_SITE: Twitter account of the site
* BLOG_TWITTER_AUTHOR: Twitter account of the post author
* BLOG_GPLUS_TYPE: Google+ Snippet type for the post object;
  (default: ``Blog``)
* BLOG_GPLUS_TYPES: Choices of available blog types for twitter;
  (default: to ``META_GPLUS_TYPES`` defined in `django-meta settings`_)
* BLOG_GPLUS_AUTHOR: Google+ account of the post author
* BLOG_ENABLE_COMMENTS: Whether to enable comments by default on posts;
  while ``djangocms_blog`` does not ship any comment system, this flag
  can be used to control the chosen comments framework; (default: ``True``)
* BLOG_USE_ABSTRACT: Use an abstract field for the post; if ``False``
  no abstract field is available for every post; (default: ``True``)
* BLOG_USE_PLACEHOLDER: Post content is managed via placeholder;
  if ``False`` a HTMLField is provided instead; (default: ``True``)
* BLOG_USE_RELATED: Enable related posts to link one post to others;
  (default: ``True``)
* BLOG_MULTISITE: Add support for multisite setup; (default: ``True``)
* BLOG_AUTHOR_DEFAULT: Use a default if not specified; if set to ``True`` the
  current user is set as the default author, if set to ``False`` no default
  author is set, if set to a string the user with the provided username is
  used; (default: ``True``)
* BLOG_DEFAULT_PUBLISHED: If posts are marked as published by default;
  (default: ``False``)
* BLOG_ADMIN_POST_FIELDSET_FILTER: Callable function to change(add or filter)
  fields to fieldsets for admin post edit form; (default: ``False``). Function example:

.. code-block:: python

    def fieldset_filter_function(fsets, request, obj=None):
        if request.user.groups.filter(name='Editor').exists():
            fsets[1][1]['fields'][0].append('author')  # adding 'author' field if user is Editor
        return fsets

* BLOG_AVAILABLE_PERMALINK_STYLES: Choices of permalinks styles;
* BLOG_PERMALINK_URLS: URLConf corresponding to
  BLOG_AVAILABLE_PERMALINK_STYLES;
* BLOG_URLCONF: Apphoo URLConf; (default: ``'djangocms_blog.urls'``);
* BLOG_DEFAULT_OBJECT_NAME: Default name for Blog item (used in django CMS Wizard);
* BLOG_AUTO_SETUP: Enable the blog **Auto setup** feature; (default: ``True``)
* BLOG_AUTO_HOME_TITLE: Title of the home page created by **Auto setup**;
  (default: ``Home``)
* BLOG_AUTO_BLOG_TITLE: Title of the blog page created by **Auto setup**;
  (default: ``Blog``)
* BLOG_AUTO_APP_TITLE: Title of the ``BlogConfig`` instance created by
  **Auto setup**; (default: ``Blog``)
* BLOG_AUTO_NAMESPACE: Instance namespace of the ``BlogConfig`` instance created by
  **Auto setup**; (default: ``Blog``)
* BLOG_SITEMAP_PRIORITY_DEFAULT: Default priority for sitemap items; (default: ``0.5``)
* BLOG_SITEMAP_CHANGEFREQ: List for available changefreqs for sitemap items; (default: **always**,
  **hourly**, **daily**, **weekly**, **monthly**, **yearly**, **never**)
* BLOG_SITEMAP_CHANGEFREQ_DEFAULT: Default changefreq for sitemap items; (default: ``monthly``)
* BLOG_CURRENT_POST_IDENTIFIER: Current post identifier in request (default ``djangocms_post_current``)
* BLOG_CURRENT_NAMESPACE: Current post config identifier in request  (default: ``djangocms_post_current_config``)
* BLOG_ENABLE_THROUGH_TOOLBAR_MENU: Is the toolbar menu throught whole all applications (default: ``False``)
* BLOG_PLUGIN_MODULE_NAME: Blog plugin module name (default: ``Blog``)
* BLOG_LATEST_ENTRIES_PLUGIN_NAME: Blog latest entries plugin name (default: ``Latest Blog Articles``)
* BLOG_AUTHOR_POSTS_PLUGIN_NAME: Blog author posts plugin name (default: ``Author Blog Articles``)
* BLOG_TAGS_PLUGIN_NAME: Blog tags plugin name (default: ``Tags``)
* BLOG_CATEGORY_PLUGIN_NAME: Blog categories plugin name (default: ``Categories``)
* BLOG_ARCHIVE_PLUGIN_NAME: Blog archive plugin name (default: ``Archive``)
* BLOG_FEED_CACHE_TIMEOUT: Cache timeout for RSS feeds
* BLOG_FEED_INSTANT_ITEMS: Number of items in Instant Article feed
* BLOG_FEED_LATEST_ITEMS: Number of items in latest items feed
* BLOG_FEED_TAGS_ITEMS: Number of items in per tags feed
* BLOG_PLUGIN_TEMPLATE_FOLDERS: (Sub-)folder from which the plugin templates are loaded. The default folder is ``plugins``. It goes into the ``djangocms_blog`` template folder (or, if set, the folder named in the app hook). This allows, e.g., different templates for showing a post list as tables, columns, ... . New templates have the same names as the standard templates in the ``plugins`` folder (``latest_entries.html``, ``authors.html``, ``tags.html``, ``categories.html``, ``archive.html``). Default behavior corresponds to this setting being ``( ("plugins", _("Default template") )``. To add new templates add to this setting, e.g., ``('timeline', _('Vertical timeline') )``.
* BLOG_META_DESCRIPTION_LENGTH: Maximum length for the Meta description field (default: ``320``)
* BLOG_META_TITLE_LENGTH: Maximum length for the Meta title field (default: ``70``)
* BLOG_ABSTRACT_CKEDITOR: Configuration for the CKEditor of the abstract field (as per https://github.com/divio/djangocms-text-ckeditor/#customizing-htmlfield-editor)
* BLOG_POST_TEXT_CKEDITOR: Configuration for the CKEditor of the post content field

******************
Read-only settings
******************

* BLOG_MENU_TYPES: Available structures of the Blog menu; (default list **Posts and Categories**,
  **Categories only**, **Posts only**, **None**)
* BLOG_MENU_TYPE: Structure of the Blog menu;
  (default: ``Posts and Categories``)


********************
Per-Apphook settings
********************

The following settings can be configured for each ``Apphook config``: the settings above will
be used as defaults.

* application title: Free text title that can be used as title in templates;
* object name: Free text label for Blog items in django CMS Wizard;
* Post published by default: Per-Apphook setting for BLOG_DEFAULT_PUBLISHED;
* Permalink structure: Per-Apphook setting for
  BLOG_AVAILABLE_PERMALINK_STYLES;
* Use placeholder and plugins for article body: Per-Apphook setting for
  BLOG_USE_PLACEHOLDER;
* Use abstract field: Per-Apphook setting for BLOG_USE_ABSTRACT;
* Enable related posts: Per-Apphook setting for BLOG_USE_RELATED;
* Set author: Per-Apphook setting for BLOG_AUTHOR_DEFAULT;
* Paginate sizePer-Apphook setting for BLOG_PAGINATION;
* Template prefix: Alternative directory to load the blog templates from;
* Menu structure: Per-Apphook setting for BLOG_MENU_TYPE
* Show empty categories in menu: Per-Apphook setting for BLOG_MENU_EMPTY_CATEGORIES
* Sitemap changefreq: Per-Apphook setting for BLOG_SITEMAP_CHANGEFREQ_DEFAULT
* Sitemap priority: Per-Apphook setting for BLOG_SITEMAP_PRIORITY_DEFAULT
* Object type: Per-Apphook setting for BLOG_TYPE
* Facebook type: Per-Apphook setting for BLOG_FB_TYPE
* Facebook application ID: Per-Apphook setting for BLOG_FB_APP_ID
* Facebook profile ID: Per-Apphook setting for BLOG_FB_PROFILE_ID
* Facebook page URL: Per-Apphook setting for BLOG_FB_PUBLISHER
* Facebook author URL: Per-Apphook setting for BLOG_AUTHOR_URL
* Facebook author: Per-Apphook setting for BLOG_AUTHOR
* Twitter type: Per-Apphook setting for BLOG_TWITTER_TYPE
* Twitter site handle: Per-Apphook setting for BLOG_TWITTER_SITE
* Twitter author handle: Per-Apphook setting for BLOG_TWITTER_AUTHOR
* Google+ type: Per-Apphook setting for BLOG_GPLUS_TYPE
* Google+ author name: Per-Apphook setting for BLOG_GPLUS_AUTHOR
* Send notifications on post publish: Send desktop notifications when a post is published
* Send notifications on post update: Send desktop notifications when a post is updated


.. _django-meta settings: https://github.com/nephila/django-meta#settings


docs/autodoc/settings.rst
.. _settings:

#############
Configuration
#############

Django settings
###############

.. automodule:: djangocms_blog.settings
    :members:
    :private-members:

Per-apphook settings
####################

.. autoclass:: djangocms_blog.cms_appconfig.BlogConfigForm
    :members:
    :private-members:


HISTORY.rst
.. :changelog:

*******
History
*******

.. towncrier release notes start

1.2.3 (2020-12-21)
==================

Features
--------

- Improve django-app-enabler support (#660)
- Update translations
- Update django-app-enabler information


1.2.2 (2020-12-20)
==================

Features
--------

- Add support for Python 3.9 (#657)


Bugfixes
--------

- Handle unicode chars in reverse of Post and Category models, using Django path() method instead of url() (#653)


Improved Documentation
----------------------

- Update docs to clarify how to add content (#636)


1.2.1 (2020-11-16)
==================

Bugfixes
--------

- Fix admin urlconf not matching path syntax (#648)


1.2.0 (2020-11-14)
==================

- BREAKING CHANGE: Replace url() function with path() (#645)

Features
--------

- Add support for django CMS 3.8 / Django 3.1 (#643)
- Update to modern tooling and port to github actions (#564)
- Add option to allow fallback language in post/category URLs (#546)
- Move post slug to top of post admin form (#567)
- Add blog post extensions (#569)
- Replace deprecated function calls (#571)
- Improve tag selection UX/UI (#614)
- Improve admin customization (#620)
- Improve documentation (#622)
- Add rtd config file and refactor test / docs dependencies (#624)
- Replace G+ metadata with Schema.org (#637)
- Update pre-commit checks to latest version (#639)


Bugfixes
--------

- Do not let site crash on menu if there is a namespace mismatch (#532)
- Improve setup documentation (#541)
- Fix GA on pull request (#575)
- Fix coveralls failure on pull request in GA (#577)
- Fix link in PR template (#581)
- Skip haystack index creation if aldryn-search is not installed but haystack is (#584)
- Skip towncrier on develop / master branch (#591)
- Fix missing template folders selection in plugins (#595)
- Clarify documentation regarding templates customization (#595)
- Fix extra migration when customizing BLOG_PLUGIN_TEMPLATE_FOLDERS (#597)
- Set default pytest configuration (#598)
- Add missing condition for (date_published) <li> on blog_meta template (#603)
- Fix python version declaration in tox (#606)
- Fix creating text plugin in wizard (#607)
- Split linting GA to its own file (#608)
- Doc improvements for usage with djangocms-page-meta (#613)
- Update linting (#618)
- Fix building docs (#632)


Improved Documentation
----------------------

- Improve documentation to enable social meta tags rendering (#613)


Misc
----

- #593


1.1.1 (2020-05-15)
==================

* Fix channels support
* Avoid admin exception for related posts when config is None
* Fix error when overriding templates folder

1.1.0 (2020-05-04)
==================

* Add support for django 3.0
* Add BlogAuthorPostsListPlugin to show posts per author
* Add experimental support for django-app-enabler
* Remove cmsplugin_filer from installation docs
* Set minimum django-parler version to 2.0
* Reduce the maximum length of slug fields to 752 characters
* Fix duplicated authors in BlogAuthorPostsPlugin
* Fix to respect current locale for category names
* Improve documentation for meta tags

1.0.0 (2019-11-04)
==================

* Add support for django CMS 3.7
* Add support for Python 3.7
* Add image size meta for Facebook
* Add support for django-parler >= 2
* Move to django-app-helper
* Drop support for Django < 1.11
* Drop support for django CMS < 3.5
* Drop older compatibilities

0.9.11 (2019-08-06)
===================

* Use menu_empty_categories config for BlogCategoryPlugin
* Purge menu cache when deleting a BlogConfig

0.9.10 (2019-07-02)
===================

* Fixed allow_unicode kwarg for AutoSlugField
* Fixed sphinx conf isort
* Set category as requested or not depending on the permalink setting

0.9.9 (2019-04-05)
==================

* Fixed issue with thumbnails not being preserved in admin form
* Pinned django-taggit version

0.9.8 (2019-01-13)
==================

* Fixed test environment in Django 1.8, 1.9
* Added related posts to templates / documentation
* Added a fix for multiple error messages when slug is not unique

0.9.7 (2018-05-05)
==================

* Fixed subtitle field not added to the admin

0.9.6 (2018-05-02)
==================

* Fixed string representation when model has no language
* Added subtitle field

0.9.5 (2018-04-07)
==================

* Fixed jquery path in Django 1.9+"Fix jquery path in Django 1.9+
* Added configurable blog abstract/text CKEditor

0.9.4 (2018-03-24)
==================

* Fixed migration error from 0.8 to 0.9

0.9.3 (2018-03-12)
==================

* Added dependency on lxml used in feeds
* Fixed warning on django CMS 3.5
* Fixed wizard in Django 1.11
* Updated translations

0.9.2 (2018-02-27)
==================

* Fixed missing migration

0.9.1 (2018-02-22)
==================

* Added Django 1.11 support

0.9.0 (2018-02-20)
==================

* Added support for django CMS 3.4, 3.5
* Dropped support for Django<1.8, django CMS<3.2.
* Added liveblog application.
* Refactored plugin filters: by default only data for current site are now shown.
* Added global and per site posts count to BlogCategory.
* Added option to hide empty categories from menu.
* Added standalone documentation at https://djangocms-blog.readthedocs.io.
* Enabled cached version of BlogLatestEntriesPlugin.
* Added plugins templateset.
* Improved category admin to avoid circular relationships.
* Dropped strict dependency on aldryn-search, haystack. Install separately for search support.
* Improved admin filtering.
* Added featured date to post.
* Fixed issue with urls in sitemap if apphook is not published
* Moved template to easy_thumbnails_tags template tag. Require easy_thumbnails >= 2.4.1
* Made HTML description and title fields length configurable
* Added meta representation for CategoryEntriesView
* Generated valid slug in wizard if the given one is taken
* Fixed error in category filtering when loading the for via POST
* Returned 404 in AuthorEntriesView if author does not exists

djangocms_blog/settings.py
"""
List of settings that can be set in project django settings.
"""
from django.utils.translation import gettext_lazy as _
from meta import settings as meta_settings

MENU_TYPE_COMPLETE = "complete"
MENU_TYPE_CATEGORIES = "categories"
MENU_TYPE_POSTS = "posts"
MENU_TYPE_NONE = "none"
DATE_FORMAT = "%a %d %b %Y %H:%M"

PERMALINK_TYPE_FULL_DATE = "full_date"
PERMALINK_TYPE_SHORT_DATE = "short_date"
PERMALINK_TYPE_CATEGORY = "category"
PERMALINK_TYPE_SLUG = "slug"

BLOG_DEFAULT_PUBLISHED = False
"""
.. _DEFAULT_PUBLISHED:

Default post published status.
"""

PERMALINKS = (  # noqa
    (PERMALINK_TYPE_FULL_DATE, _("Full date")),
    (PERMALINK_TYPE_SHORT_DATE, _("Year /  Month")),
    (PERMALINK_TYPE_CATEGORY, _("Category")),
    (PERMALINK_TYPE_SLUG, _("Just slug")),
)
"""
.. _PERMALINKS:

Permalinks styles.
"""

PERMALINKS_URLS = {  # noqa
    PERMALINK_TYPE_FULL_DATE: "<int:year>/<int:month>/<int:day>/<str:slug>/",
    PERMALINK_TYPE_SHORT_DATE: "<int:year>/<int:month>/<str:slug>/",
    PERMALINK_TYPE_CATEGORY: "<str:category>/<str:slug>/",
    PERMALINK_TYPE_SLUG: "<str:slug>/",
}

"""
.. _PERMALINKS_URLS:

Permalinks urlconfs.
"""

MENU_TYPES = (  # noqa
    (MENU_TYPE_COMPLETE, _("Categories and posts")),
    (MENU_TYPE_CATEGORIES, _("Categories only")),
    (MENU_TYPE_POSTS, _("Posts only")),
    (MENU_TYPE_NONE, _("None")),
)
"""
.. _DEFAULT_MENU_TYPES:

Types of menu structure.
"""

SITEMAP_CHANGEFREQ_LIST = (  # noqa
    ("always", _("always")),
    ("hourly", _("hourly")),
    ("daily", _("daily")),
    ("weekly", _("weekly")),
    ("monthly", _("monthly")),
    ("yearly", _("yearly")),
    ("never", _("never")),
)
"""
.. _SITEMAP_CHANGEFREQ_LIST:

List of changefreqs defined in sitemaps.
"""

BLOG_IMAGE_THUMBNAIL_SIZE = {"size": "120x120", "crop": True, "upscale": False}
"""
.. _IMAGE_THUMBNAIL_SIZE:

Easy-thumbnail alias configuration for the post main image when shown on the post lists;
it's a dictionary with ``size``, ``crop`` and ``upscale`` keys.
"""

BLOG_IMAGE_FULL_SIZE = {"size": "640", "crop": True, "upscale": False}
"""
.. _IMAGE_FULL_SIZE:

Easy-thumbnail alias configuration for the post main image when shown on the post detail;
it's a dictionary with ``size``, ``crop`` and ``upscale`` keys.
"""

BLOG_URLCONF = (
    ("djangocms_blog.urls", _("Blog: Blog list at root url of blog")),
    ("djangocms_blog.urls_hub", _("Content hub: Category list at root url of blog")),
)
"""
.. _URLCONF:

List of alternative URL configurations which can be set per app hook.
"""

BLOG_PAGINATION = 10
"""
.. _PAGINATION:

Number of post per page.
"""

BLOG_LATEST_POSTS = 5
"""
.. _LATEST_POSTS:

Default number of post in the **Latest post** plugin.
"""

BLOG_POSTS_LIST_TRUNCWORDS_COUNT = 100
"""
.. _POSTS_LIST_TRUNCWORDS_COUNT:

Default number of words shown for abstract in the post list.
"""

BLOG_META_DESCRIPTION_LENGTH = 320
"""
.. _META_DESCRIPTION_LENGTH:

Maximum length for the Meta description field.
"""

BLOG_META_TITLE_LENGTH = 70
"""
.. _META_TITLE_LENGTH:

Maximum length for the Meta title field.
"""

BLOG_MENU_TYPES = MENU_TYPES
"""
.. _MENU_TYPES:

List of available menu types.
"""

BLOG_MENU_EMPTY_CATEGORIES = True
"""
.. _MENU_EMPTY_CATEGORIES:

Flag to show / hide categories without posts attached from the menu.
"""

BLOG_TYPE = "Article"
"""
.. _TYPE:

Generic type for the post object.
"""

BLOG_TYPES = meta_settings.OBJECT_TYPES
"""
.. _TYPES:

Choices of available blog types.

Available values are defined in to ``META_OBJECT_TYPES`` defined in `django-meta settings`_.
"""

BLOG_FB_TYPE = "Article"
"""
.. _FB_TYPE:

Open Graph type for the post object.
"""

BLOG_FB_TYPES = meta_settings.FB_TYPES
"""
.. _FB_TYPES:

Choices of available blog types.

Available values are defined in to ``META_FB_TYPES`` defined in `django-meta settings`_.
"""

BLOG_FB_APPID = meta_settings.FB_APPID
"""
.. _FB_APPID:

Facebook Application ID.

Default from ``FB_APPID`` defined in `django-meta settings`_.
"""

BLOG_FB_PROFILE_ID = meta_settings.FB_PROFILE_ID
"""
.. _FB_PROFILE_ID:

Facebook profile ID of the post author.

Default from ``FB_PROFILE_ID`` defined in `django-meta settings`_.
"""

BLOG_FB_PUBLISHER = meta_settings.FB_PUBLISHER
"""
.. _FB_PUBLISHER:

Facebook URL of the blog publisher.

Default from ``FB_PUBLISHER`` defined in `django-meta settings`_.
"""

BLOG_FB_AUTHOR_URL = "get_author_url"
"""
.. _FB_AUTHOR_URL:

Facebook profile URL of the post author.
"""

BLOG_FB_AUTHOR = "get_author_name"
"""
.. _FB_AUTHOR:

Facebook profile URL of the post author.
"""

BLOG_TWITTER_TYPE = "summary"
"""
.. _TWITTER_TYPE:

Twitter Card type for the post object.
"""

BLOG_TWITTER_TYPES = meta_settings.TWITTER_TYPES
"""
.. _TWITTER_TYPES:

Choices of available blog types for twitter.

Default from ``TWITTER_TYPES`` defined in `django-meta settings`_.
"""

BLOG_TWITTER_SITE = meta_settings.TWITTER_SITE
"""
.. _TWITTER_SITE:

Twitter account of the site.

Default from ``TWITTER_SITE`` defined in `django-meta settings`_.
"""

BLOG_TWITTER_AUTHOR = "get_author_twitter"
"""
.. _TWITTER_AUTHOR:

Twitter account of the post author.
"""

BLOG_SCHEMAORG_TYPE = "Blog"
"""
.. _SCHEMAORG_TYPE:

Schema.org type for the post object.
"""

BLOG_SCHEMAORG_TYPES = meta_settings.SCHEMAORG_TYPES
"""
.. _SCHEMAORG_TYPES:

Choices of available Schema.org types.

Default from ``SCHEMAORG_TYPES`` defined in `django-meta settings`_.
"""

BLOG_SCHEMAORG_AUTHOR = "get_author_schemaorg"
"""
.. _SCHEMAORG_AUTHOR:

Google+ account of the post author (deprecated).
"""

BLOG_ENABLE_COMMENTS = True
"""
.. _ENABLE_COMMENTS:

Whether to enable comments by default on posts

While ``djangocms_blog`` does not ship any comment system, this flag
can be used to control the chosen comments framework.
"""

BLOG_USE_ABSTRACT = True
"""
.. _USE_ABSTRACT:

Use an abstract field for the post.

If ``False`` no abstract field is available for posts.
"""

BLOG_USE_PLACEHOLDER = True
"""
.. _USE_PLACEHOLDER:

Post content is managed via placeholder

If ``False`` a HTMLField is provided instead.
"""

BLOG_USE_RELATED = True
"""
.. _USE_RELATED:

Enable related posts to link one post to others.
"""

BLOG_MULTISITE = True
"""
.. _MULTISITE:

Add support for multisite setup.
"""

BLOG_AUTHOR_DEFAULT = True
"""
.. _AUTHOR_DEFAULT:

Use a default if not specified:

* ``True``: the current user is set as the default author;
* ``False``: no default author is set;
* any other string: the user with the provided username is used;
"""

BLOG_ADMIN_POST_FIELDSET_FILTER = False
"""
.. _ADMIN_POST_FIELDSET_FILTER:

Callable function to change (add or filter) fields to fieldsets for admin post edit form.

See :ref:`admin_filter_function` for more details.
"""

BLOG_AVAILABLE_PERMALINK_STYLES = PERMALINKS
"""
.. _AVAILABLE_PERMALINK_STYLES:

Choices of permalinks styles.
"""

BLOG_PERMALINK_URLS = PERMALINKS_URLS
"""
.. _PERMALINK_URLS:

URLConf corresponding to :ref:`BLOG_AVAILABLE_PERMALINK_STYLES <AVAILABLE_PERMALINK_STYLES>`.
"""

BLOG_DEFAULT_OBJECT_NAME = "Article"
"""
.. _DEFAULT_OBJECT_NAME:

Default label for Blog item (used in django CMS Wizard).
"""

BLOG_AUTO_SETUP = True
"""
.. _AUTO_SETUP:

Enable the blog :ref:`auto_setup` feature.
"""

BLOG_AUTO_HOME_TITLE = "Home"
"""
.. _AUTO_HOME_TITLE:

Title of the home page created by :ref:`auto_setup`.
"""

BLOG_AUTO_BLOG_TITLE = "Blog"
"""
.. _AUTO_BLOG_TITLE:

Title of the blog page created by :ref:`auto_setup`.
"""

BLOG_AUTO_APP_TITLE = "Blog"
"""
.. _AUTO_APP_TITLE:

Title of the ``BlogConfig`` instance created by :ref:`auto_setup`.
"""

BLOG_AUTO_NAMESPACE = "Blog"
"""
.. _AUTO_NAMESPACE:

Namespace of the ``BlogConfig`` instance created by :ref:`auto_setup`.
"""

BLOG_SITEMAP_PRIORITY_DEFAULT = "0.5"
"""
.. _SITEMAP_PRIORITY_DEFAULT:

Default priority for sitemap items.
"""

BLOG_SITEMAP_CHANGEFREQ = SITEMAP_CHANGEFREQ_LIST
"""
.. _SITEMAP_CHANGEFREQ:

List for available changefreqs for sitemap items.
"""

BLOG_SITEMAP_CHANGEFREQ_DEFAULT = "monthly"
"""
.. _SITEMAP_CHANGEFREQ_DEFAULT:

Default changefreq for sitemap items.
"""

BLOG_ABSTRACT_CKEDITOR = True
"""
.. _ABSTRACT_CKEDITOR:

Configuration for the CKEditor of the abstract field.

See https://github.com/divio/djangocms-text-ckeditor/#customizing-htmlfield-editor for details.
"""

BLOG_POST_TEXT_CKEDITOR = True
"""
.. _POST_TEXT_CKEDITOR:

Configuration for the CKEditor of the post content field.

See https://github.com/divio/djangocms-text-ckeditor/#customizing-htmlfield-editor for details.
"""

BLOG_ENABLE_SEARCH = True
"""
.. _ENABLE_SEARCH:

Enable ``aldryn-search`` (i.e.: ``django-haystack``) indexes.
"""

BLOG_CURRENT_POST_IDENTIFIER = "djangocms_post_current"
"""
.. _CURRENT_POST_IDENTIFIER:

Current post identifier in request.

Name of the request attribute used in :py:class:`djangocms_blog.cms_toolbars.BlogToolbar` to detect if
request match a post detail.
"""

BLOG_CURRENT_NAMESPACE = "djangocms_post_current_config"
"""
.. _CURRENT_NAMESPACE:

Current post config identifier in request.

Name of the request attribute used in :py:class:`djangocms_blog.cms_toolbars.BlogToolbar` to detect the
current apphook namespace.
"""

BLOG_ENABLE_THROUGH_TOOLBAR_MENU = False
"""
.. _ENABLE_THROUGH_TOOLBAR_MENU:

Show djangocms-blog toolbar in any page, even when outside the blog apphooks.
"""

BLOG_PLUGIN_MODULE_NAME = _("Blog")
"""
.. _PLUGIN_MODULE_NAME:

Name of the djangocms-blog plugins module (group).
"""

BLOG_LATEST_ENTRIES_PLUGIN_NAME = _("Latest Blog Articles")
"""
.. _LATEST_ENTRIES_PLUGIN_NAME:

Name of the plugin showing the list of latest posts.
"""

BLOG_LATEST_ENTRIES_PLUGIN_NAME_CACHED = _("Latest Blog Articles - Cache")
"""
.. _LATEST_ENTRIES_PLUGIN_NAME_CACHED:

Name of the plugin showing the list of latest posts (cached version).
"""

BLOG_AUTHOR_POSTS_PLUGIN_NAME = _("Author Blog Articles")
"""
.. _AUTHOR_POSTS_PLUGIN_NAME:

Name of the plugin showing the list of blog posts authors.
"""

BLOG_AUTHOR_POSTS_LIST_PLUGIN_NAME = _("Author Blog Articles List")
"""
.. _AUTHOR_POSTS_LIST_PLUGIN_NAME:

Name of the plugin showing the list of posts per authors.
"""

BLOG_TAGS_PLUGIN_NAME = _("Tags")
"""
.. _TAGS_PLUGIN_NAME:

Name of the plugin showing the tag blog cloud.
"""

BLOG_CATEGORY_PLUGIN_NAME = _("Categories")
"""
.. _CATEGORY_PLUGIN_NAME:

Name of the plugin showing the list of blog categories.
"""

BLOG_ARCHIVE_PLUGIN_NAME = _("Archive")
"""
.. _ARCHIVE_PLUGIN_NAME:

Name of the plugin showing the blog archive index.
"""

BLOG_FEED_CACHE_TIMEOUT = 3600
"""
.. _FEED_CACHE_TIMEOUT:

Cache timeout for RSS feeds.
"""

BLOG_FEED_INSTANT_ITEMS = 50
"""
.. _FEED_INSTANT_ITEMS:

Number of items in Instant Article feed.
"""

BLOG_FEED_LATEST_ITEMS = 10
"""
.. _FEED_LATEST_ITEMS:

Number of items in latest items feed.
"""

BLOG_FEED_TAGS_ITEMS = 10
"""
.. _FEED_TAGS_ITEMS:

Number of items in per tags feed.
"""

BLOG_LIVEBLOG_PLUGINS = ("LiveblogPlugin",)
"""
.. _LIVEBLOG_PLUGINS:

List of plugins implementing the :ref:`liveblog` feature.
"""

BLOG_PLUGIN_TEMPLATE_FOLDERS = (("plugins", _("Default template")),)
"""
.. _PLUGIN_TEMPLATE_FOLDERS:

(Sub-)folder from which the plugin templates are loaded.

The default folder is ``plugins``.

See :ref:`plugin_templates` for more details.
"""

BLOG_USE_FALLBACK_LANGUAGE_IN_URL = False
"""
.. _USE_FALLBACK_LANGUAGE_IN_URL:

When displaying URL, prefer URL in the fallback language if an article or category is not available in the
current language.
"""

BLOG_WIZARD_CONTENT_PLUGIN = "TextPlugin"
"""
.. _WIZARD_CONTENT_PLUGIN:

Name of the plugin created by wizard for the text content.
"""

BLOG_WIZARD_CONTENT_PLUGIN_BODY = "body"
"""
.. _WIZARD_CONTENT_PLUGIN_BODY:

Name of the plugin field to add wizard text.
"""

BLOG_STRUCTURE = 0
"""
.. _STRUCTURE:

Default structure of blog: 0 for a list of posts ordered by publication date. 1 for a set of categories ordered by
priority.
"""

params = {param: value for param, value in locals().items() if param.startswith("BLOG_")}

"""
.. _django-meta settings: https://github.com/nephila/django-meta#settings
"""


def get_setting(name):
    """Get setting value from django settings with fallback to globals defaults."""
    from django.conf import settings

    return getattr(settings, "BLOG_%s" % name, params["BLOG_%s" % name])
